libs/blocks/comparison-table/comparison-table.js

```diff
@@ -0,0 +1,498 @@
+  const selectedIndices = new Set(visibleSelects.map((s) => +s.value));
+
+  visibleSelects.forEach((selectItem) => {
+    const currentValue = +selectItem.value;
+    selectItem.innerHTML = '';
+
+    headerTitles.forEach((title, index) => {
+      if (!title || (selectedIndices.has(index) && index !== currentValue)) return;
+      const option = createTag('option', { value: index }, title);
+      if (index === currentValue) option.selected = true;
+      selectItem.appendChild(option);
+    });
+  });
+}
+
+function handleSelectChange(e, { headerItemIndex, el, headerTitles }) {
+  const newValue = +e.target.value;
+  const isFirstVisible = headerItemIndex === getFirstVisibleColumnIndex(el);
+
+  el.querySelectorAll(`[data-column-index="${headerItemIndex}"]`).forEach((col) => col.classList.add('hidden'));
+  el.querySelectorAll(`[data-column-index="${newValue}"]`).forEach((col) => {
+    col.classList.remove('hidden');
+    const parent = col.parentNode;
+    if (!isFirstVisible || !parent) return;
+    const firstHeaderItem = parent.querySelector('.header-item:first-child');
+    if (col.classList.contains('header-item') && firstHeaderItem !== col) {
+      parent.insertBefore(col, firstHeaderItem.nextSibling);
+      return;
+    }
+    const rowHeader = parent.querySelector('.table-row-header');
+    if (rowHeader) parent.insertBefore(col, rowHeader.nextSibling);
+  });
+
+  const selectElement = el.querySelector(`[data-column-index="${newValue}"] .mobile-filter-select`);
+  if (selectElement) selectElement.value = newValue;
+  updateVisibleSelects({ el, headerTitles });
+  syncAccessibilityHeaders(el);
+}
+
+function createMobileFilterSelect({ headerTitles, headerItemIndex, el }) {
+  const select = createTag('select', { class: 'mobile-filter-select', name: 'column-filter' });
+  headerTitles.forEach((title, index) => {
+    if (!title
+      || (headerItemIndex === 1 && index === 2)
+      || (headerItemIndex === 2 && index === 1)) return;
+    const option = createTag('option', { value: index }, title);
+    if (index === headerItemIndex) option.selected = true;
+    select.appendChild(option);
+  });
+  select.addEventListener('change', (e) => handleSelectChange(e, { headerItemIndex, el, headerTitles }));
+  return select;
+}
+
+function addLastContainerElements(container) {
+  const actionAreaElements = container.querySelectorAll('.action-area');
+  if (actionAreaElements.length) {
+    const btnContainer = createTag('div', { class: 'btn-container' });
+    if (actionAreaElements.length > 1) btnContainer.classList.add('has-multiple');
+    actionAreaElements.forEach((element) => btnContainer.appendChild(element));
+    container.appendChild(btnContainer);
+  }
+  const description = container.querySelector('p:not(.action-area)');
+  (description || container.prepend(createTag('p', { class: 'description' })) || container.firstChild).classList.add('description');
```

Comment:
this is clever, but I think it would be cleaner to make the code more explicit and make the boolean chain "pure", to avoid any potential side effects, especially if someone else touches this code later down the line.